### DISCUSSION TITLE
Google&GitHub Login in Vercel-Deployment

### DISCUSSION THREAD

**Discussion by RichardBeindorf** [neuefische/questions#839] [Web]
Hello yall,

as the many students before me i ran into issues in during the vercel deployment of my github & google login feature.

I have tried to follow as many previous guidelines and tipps as possible but can't come up with a solution that works.

First i introduce you to the immediate problem:

<img width="615" alt="Bildschirmfoto 2024-05-17 um 15 28 01" src="https://github.com/neuefische/questions/assets/159543456/245f6d05-bf45-4c3b-8d86-1cf9e017ba9c">

 This is what happens, when i click on "sign in".
 
<img width="692" alt="Bildschirmfoto 2024-05-17 um 15 29 03" src="https://github.com/neuefische/questions/assets/159543456/9c3b814a-0d49-4b16-9a70-f0fc229f5a29">

Here you can see the URL of the current deployment and the actual login button. Needles to mention, all of the above works fine when used in local development.

As of for for all the tiny steps needed to successfully enable the login functionality:

1. I added all necessary environment variables to Vercel including the current PR-deployment URL like this: AUTH_URL=https://activities-h3bneb55k-volker-boehms-projects.vercel.app (this part "h3bneb55k" changes after every push)
2. I added the above mentioned URL in Form of a new Application to GitHub
<img width="770" alt="Bildschirmfoto 2024-05-17 um 15 34 53" src="https://github.com/neuefische/questions/assets/159543456/3bec2985-4a0c-4401-b763-b483bfe7831e">

3. I added the above mentioned URL as subdomain to google console
<img width="545" alt="Bildschirmfoto 2024-05-17 um 15 35 56" src="https://github.com/neuefische/questions/assets/159543456/2dae41ba-fc0a-4130-93ea-d4216c8a55fa">

4. I added the above mentioned URL to the google application domain:
<img width="891" alt="Bildschirmfoto 2024-05-17 um 15 37 05" src="https://github.com/neuefische/questions/assets/159543456/3bbcdc35-1172-4864-bc2b-80a716d414c6">

5. Since it already worked local, the api / provider route should be flawless:
<img width="491" alt="Bildschirmfoto 2024-05-17 um 15 39 43" src="https://github.com/neuefische/questions/assets/159543456/bca41ae5-8f6d-49d7-b7c3-1b66e028a8b6">

6. I have emptied by connection sessions in google.

I have been following all advices by the docs for naming conventions.
My feeling is, that the error occurs around the URL im using for deployment and the whole process around vercel since im a little clueless about that.

<img width="870" alt="Bildschirmfoto 2024-05-17 um 15 44 08" src="https://github.com/neuefische/questions/assets/159543456/5c9aab4b-f131-43c8-b962-d8132714e5c6">

I'm picking up the URL for the deployment at this spot above. I hope that is correct. I got confused about another URL deployment link provided by this Preview button.

<img width="829" alt="Bildschirmfoto 2024-05-17 um 15 45 49" src="https://github.com/neuefische/questions/assets/159543456/82ae9035-6b50-4668-87dc-16d4ba959d01">

This link always gives me the following "URL: https://activities-app-git-feature-googlelogin-volker-boehms-projects.vercel.app/"
I have also tried the whole process unsuccessfully with that URL.

Thank you for any help in advance!

Regards
Richard

**Comment by F-Kirchhoff** [comment]
You can disable the Github and Google Auth when you are in the "preview" environment of Vercel. This prevents you from having to update the two OAuth Apps every time you create a new branch. You can disable them as follows: 

```js
//[...nextauth].js

providers: [
  process.env.VERCEL_ENV !== 'preview' && GithubProvider(...),
  process.env.VERCEL_ENV !== 'preview' && GoogleProvider(...),
}

**Reply by RichardBeindorf** [reply]
I have implemented your solution and used some hard coded credentails but im still running into the error message from the start of my post! :/

**Comment by RichardBeindorf** [comment]
I have finally figured it out.
The important thing i missed was commented in the [nextauth documentation](https://next-auth.js.org/deployment): you should not include the NEXTAUTH_URL or AUTH_URL in the environment variables, opposed to the local deployment.
Also i did change the naming from NEXT_AUTH to NEXTAUTH, i didn't error test for that specificly.
